corelib-schemaorg/src/main/java/eu/europeana/corelib/edm/utils/SchemaOrgTypeFactory.java

```diff
@@ -1,187 +1,190 @@
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.corelib.solr.entity.ConceptImpl;
+import eu.europeana.corelib.solr.entity.ProxyImpl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Factory class for creating a new schema.org object from an EDM FulLBean
+ */
+public final class SchemaOrgTypeFactory {
+    private static final Map<String, String> conceptTypes;
+
+    static {
+        conceptTypes = new HashMap<>();
+        conceptTypes.put("http://data.europeana.eu/concept/base/6", SchemaOrgConstants.TYPE_BOOK);
+        conceptTypes.put("http://data.europeana.eu/concept/base/43", SchemaOrgConstants.TYPE_MAP);
+        conceptTypes.put("http://data.europeana.eu/concept/base/47", SchemaOrgConstants.TYPE_PAINTING);
+        conceptTypes.put("http://data.europeana.eu/concept/base/48", SchemaOrgConstants.TYPE_PHOTOGRAPH);
+        conceptTypes.put("http://data.europeana.eu/concept/base/51", SchemaOrgConstants.TYPE_SCULPTURE);
+        conceptTypes.put("http://data.europeana.eu/concept/base/190", SchemaOrgConstants.TYPE_VISUAL_ARTWORK);
+        conceptTypes.put("http://data.europeana.eu/concept/base/18", SchemaOrgConstants.TYPE_NEWSPAPER);
+    }
+
+    private SchemaOrgTypeFactory() {}
+
+    public static MediaObject createMediaObject(String mimeType) {
+        if (mimeType == null) {
+            return new MediaObject();
+        }
+
+        if (mimeType.startsWith("image/")) {
+            return new ImageObject();
+        }
+        if (mimeType.startsWith("audio/") || mimeType.startsWith("sound/")) {
+            return new AudioObject();
+        }
+        if (mimeType.startsWith("video/")) {
+            return new VideoObject();
+        }
+        return new MediaObject();
+    }
+
+    public static eu.europeana.corelib.edm.model.schemaorg.ContextualEntity createAgent(Agent agent) {
+        if (agent == null) {
+            return null;
+        }
+
+        if (agent.getRdaGr2DateOfEstablishment() != null || agent.getRdaGr2DateOfTermination() != null) {
+            //just Group of Persons, not EDM Organization
+            return new Organization();
+        }
+        return new Person();
+    }
+    
+    public static ContextualEntity createContextualEntity(ContextualClass entity) {
+	ContextualEntity thingObject;
+	if(entity instanceof eu.europeana.corelib.definitions.edm.entity.Organization) {
+	    thingObject = new EdmOrganization();
+	}
+	else if(entity instanceof eu.europeana.corelib.definitions.edm.entity.Concept) {
+	    thingObject = new Concept();
+	}
+	else if(entity instanceof eu.europeana.corelib.definitions.edm.entity.Agent) {
+	    thingObject = createAgent((Agent) entity);
+	}
+	else if(entity instanceof eu.europeana.corelib.definitions.edm.entity.Place) {
+	    thingObject = new Place();
+	}
+	else if(entity instanceof eu.europeana.corelib.definitions.edm.entity.Timespan) {
+	    thingObject = new Timespan();

```

Comment:
according to the current specifications Timespans are mapped to schamorg.Thing class, please update

Reply:
According to the previous comments, since the solr Timespan entity has its own fields in addition to the ones in the Thing class, I have kept it in case someone wants to add new fields in the future. However the current mapping is only for the specified fields, i.e. the ones contained in the Thing class.

Reply:
there is no Timespan type in schema.org, the decision from specifications was to use the Thing class https://schema.org/docs/search_results.html?q=thing

Reply:
please remove this class

Reply:
the class is, as agreed, kept but with no fields in addition to the Thing class.